spec.md

```diff
@@ -96,7 +96,63 @@ and eventually have it accepted as a Unicode Technical Standard (UTS).
 
 ## Data Model
 
-> _The data-only representation of a message._
+A Message provides the representation of a single message.
+It takes one of two forms,
+either as a PatternMessage or a SelectMessage.
+In a multi-locale context,
+a single message's choice of representation as one or the other
+should not affect its representation in other languages or locales.
+
+```ts
+type Message = PatternMessage | SelectMessage
+```
+
+A PatternMessage contains a list of PatternElement values,
+some of which are directly defined literal values,
+while others are placeholders with formatted values that depend on additional data.
+
+```ts
+interface PatternMessage {
+  type: 'message'
+  value: PatternElement[]
+  meta?: Meta
+}
+```
+
+SelectMessage provides for the selection of one list of PatternElement values
+to use as the message's value when formatting,
+depending on the value of one or more Selector values.
+
+Each SelectCase is defined by a `key` of one or more string identifiers,
+and selection between them is made according to the corresponding Selector values.
+From this it follows that a valid SelectMessage must have at least as many `select` entries
+as its highest count of SelectCase `key` entries within its `cases`.
+The `fallback` value of a Selector is used in addition to its `value`
+when selecting one of the `cases` during formatting.
+It should match exactly one of the corresponding SelectCase `key` values.
```

Comment:
`nit`: `should` is a loaded and ephemeral word. Are we ok if it doesn't? What then? Can it match two? What then?

Reply:
I would welcome a suggestion for clarifying this. The intent here is to indicate that with e.g. your earlier example:
```js
{
  type: 'select',
  select: [
    { value: ..., fallback: "masculine"},
    { value: ..., fallback: "few"}
  ],
  cases: [
    { value: ..., key: ["masculine", "one"] },
    { value: ..., key: ["masculine", "few"] },
    { value: ..., key: ["feminine", "one"] },
    { value: ..., key: ["feminine", "many"] }
  ]
}
```
The value of `select[0].fallback` should be either `"masculine"` or `"feminine"`, while `select[1].value` should to be one of `"one"`, `"few"`, `"many"`.

My preference here would be for the spec to use the term "should" rather than "must", as disobeying the latter would probably mean that we'd need to consider it a parsing error. I would prefer that be avoided, as it would mean that we'd need to build in relatively strong guarantees to rather remote parts of the toolchain to be able to also guarantee that result in its processing.

On the other hand, my preference here isn't absolute by any means, and I would welcome arguments to the contrary.

Reply:
I guess my question is - is it a parsing error or not?

And if not, what happens when the condition is not met?

Reply:
My position is that it should not be considered a parsing error, but instead result in all of the following:
- The default/recommended lint/validation rules should complain about it.
- During case selection, if no case matches the given selectors, the result is an empty list of pattern elements along with metadata indicating that case selection failed.
- During formatting, the representation of that implicit empty result may be determined by e.g. a `requireCaseMatch` option given to the formatter: if `false`, this is not an error, and the `formatToString()` result is en empty string. If `true`, this is an error, and some fallback representation of the message is used instead. It should be valid for a formatter to not provide this option, and always follow the `true` logic.

Reply:
I'm comfortable with moving forward with this approach for now and I'll likely suggest refinements later.